server/src/main/java/org/candlepin/dto/api/v1/NestedEntitlementTranslator.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2009 - 2020 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.dto.api.v1;
+
+import org.candlepin.dto.ModelTranslator;
+import org.candlepin.dto.ObjectTranslator;
+import org.candlepin.model.Entitlement;
+import org.candlepin.util.Util;
+
+/**
+ * This class provides translation from {@link Entitlement} model objects
+ * to {@link NestedEntitlementDTO}.
+ */
+public class NestedEntitlementTranslator implements ObjectTranslator<Entitlement, NestedEntitlementDTO> {
```

Comment:
Since NestedEntitlementDTO has only 2 fields: id and href, and it is only used in PoolDTO currently (no other DTO has an EntitlementDTO member field), the translator might be a bit overkill, and we could get away to just creating an object on the fly and calling the 2 setters, just like you did with ProvidedProductDTO.

Reply:
Done.